File: runtime/python/lambdalib.py
# Python runtime for the Lambda Calculus with floats.

import math

f_true         = (lambda a : lambda b : a)
f_false        = (lambda a : lambda b : b)
f_to_church    = (lambda n : lambda f : lambda a : a  if n < 1 else f(((f_to_church(n - 1))(f))(a)))
f_add          = (lambda a : lambda b : a+b)
f_sub          = (lambda a : lambda b : a-b)
f_mul          = (lambda a : lambda b : a*b)
f_div          = (lambda a : lambda b : a/b)
f_mod          = (lambda a : lambda b : a%b)
f_log          = (lambda a : lambda b : math.log(b)/math.log(a))
f_pow          = (lambda a : lambda b : math.pow(a,b))
f_less_than    = (lambda a : lambda b : f_true if a < b else f_false)
f_greater_than = (lambda a : lambda b : f_true if a > b else f_false)
f_equal        = (lambda a : lambda b : f_true if a == b else f_false)
f_foor         = (lambda a : math.floor(a))
f_ceil         = (lambda a : math.ceil(a))
f_round        = (lambda a : round(a))

File: runtime/javascript/lambdalib.js
// JavaScript runtime for the Lambda Calculus with floats.

f_true = function(a){return function(b){return a}}
f_false = function(a){return function(b){return b}}
f_to_church = (function (n) { return (function (f) { return (function (a) { for (var i=0; i<n; ++i) a = f(a); return a; }) }) })
f_from_church = (function (n) {return ((n(function (a) {return (a + 1)}))(0))})
f_add = (function (a) {return (function (b) {return (a + b)})})
f_mul = (function (a) {return (function (b) {return (a * b)})})
f_div = (function (a) {return (function (b) {return (a / b)})})
f_floor = (function (a) {return Math.floor(a)})
f_ceil = (function (a) {return Math.ceil(a)})
f_equal = (function (a) {return (function (b) {return a === b ? f_true : f_false})})
f_less_than = (function (a) {return (function (b) {return a < b ? f_true : f_false})})
f_greater_than = (function (a) {return (function (b) {return a > b ? f_true : f_false})})
f_mod = (function (a) {return (function (b) {return (a % b)})})
f_log = (function (a) {return (function (b) {return Math.log(b) / Math.log(a)})})
f_pow = (function (a) {return (function (b) {return Math.pow(a,b)})})
